api/router/exec.go

```diff
@@ -73,10 +56,87 @@ func (e *ExecRouter) ExecCreateHandler(w http.ResponseWriter, r *http.Request) {
 	userID := middleware.GetUserIDFromContext(ctx)
 	projectID := middleware.GetProjectIDFromContext(ctx)
 
-	exec, err := e.service(scr.Provider).Create(ctx, projectID, userID, *scr)
+	exec, err := e.service.Create(ctx, projectID, userID, *scr)
 	if err != nil {
 		render.Render(w, r.WithContext(ctx), types.ErrHTTPError(err, "Failed to create session"))
 		return
 	}
 	render.JSON(w, r, exec)
 }
+
+func (e *ExecRouter) ExecGetHandler(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	log.Ctx(ctx).Info().Msgf("Executing ExecGet request")
+
+	execID := chi.URLParam(r, "exec")
+	if execID == "" {
+		err := fmt.Errorf("missing execID")
+		render.Render(w, r.WithContext(ctx), types.ErrHTTPBadRequest(err, "Invalid request"))
+		return
+	}
+
+	exec, err := e.store.Get(execID)
+	if err != nil {
+		render.Render(w, r.WithContext(ctx), types.ErrHTTPError(err, "Failed to get session"))
+		return
+	}
+	render.JSON(w, r, exec)
+}
+
+func (e *ExecRouter) ExecListHandler(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	log.Ctx(ctx).Info().Msgf("Executing ExecList request")
+
+	listTerminated := r.URL.Query().Get("terminated") == "true"
+	projectID := middleware.GetProjectIDFromContext(ctx)
+
+	execs, err := e.service.List(ctx, projectID)
+	if err != nil {
+		render.Render(w, r.WithContext(ctx), types.ErrHTTPError(err, "Failed to list sessions"))
+		return
+	}
+
+	if listTerminated {
+		render.JSON(w, r, types.ExecsListResponse{Execs: execs})
+		return
+	}
+
+	var res []types.Exec
+
+	for _, exec := range execs {
+		if exec.Status == types.StatusTerminated ||
+			exec.Status == types.StatusError ||
+			exec.Status == types.StatusFailed {
+			continue
+		}
+		res = append(res, exec)
+	}
+
+	render.JSON(w, r, types.ExecsListResponse{Execs: res})
+}
+
+func (e *ExecRouter) ExecTerminateHandler(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	log.Ctx(ctx).Info().Msgf("Executing ExecTerminate request")
+
+	execID := chi.URLParam(r, "exec")
+	if execID == "" {
+		err := fmt.Errorf("missing execID")
+		render.Render(w, r.WithContext(ctx), types.ErrHTTPBadRequest(err, "Invalid request"))
+		return
+	}
+
+	_, err := e.service.Get(ctx, execID)
+	if err != nil {
+		render.Render(w, r.WithContext(ctx), types.ErrHTTPError(err, "Failed to get session"))
+		return
+	}
+
+	err = e.service.Terminate(ctx, execID)
+	if err != nil {
+		render.Render(w, r.WithContext(ctx), types.ErrHTTPError(err, "Failed to terminate session"))
+		return
+	}
+
+	render.Status(r, http.StatusOK)
+}
```

Comment:
some of this needs to be factored to use the new router in the exec service but i think it's fine for now to just get things that work merged in